File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode createBST(int[] nums,int si,int ei) {
        //Base Case 
        if(si > ei) return null;
        //Calculate Mid idx
        int mid = (si+ei)/2;
        TreeNode root = new TreeNode(nums[mid]);
        //Recursive Call to capture left and right create BSTs.
        root.left = createBST(nums,si,mid-1);
        root.right = createBST(nums,mid+1,ei);
        
        return root;
    }
    public TreeNode sortedArrayToBST(int[] nums) {
        return createBST(nums,0,nums.length-1);
    }
}

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* makeBST(vector<int>& nums,int si,int ei) {
        if(si>ei) return nullptr;
        
        int mid = (si+ei)/2;
        TreeNode* root = new TreeNode(nums[mid]);
        root->left = makeBST(nums,si,mid-1);
        root->right = makeBST(nums,mid+1,ei);
        
        return root;
    }
    
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return makeBST(nums,0,nums.size()-1);
    }
};